### PR TITLE
test: Adding test for new override feature of analysis-core.

### PR DESCRIPTION
**Pull request by benjaminfuchs**
See: https://github.com/jenkinsci/analysis-core-plugin/pull/45

@uhafner

### CODE REVIEW

**Comment by olivergondza** [comment]
I was a bit too quick merging this one neglecting the mentions, sorry.

BTW, is it ok for the tests to pass if the feature it covers was not released yet? That is way I understand the chain of PRs at least ...

**Comment by benjaminfuchs** [comment]
It seems the test is not enabled in the jenkins job.

When I run locally following it fails:
`JENKINS_VERSION=1.642.3 BROWSER=chrome mvn -Dtest=WarningsPluginTest#should_override_tab_name_with_groovy_parser`

And this passes:
`LOCAL_SNAPSHOTS=true JENKINS_VERSION=1.642.3 BROWSER=chrome mvn -Dtest=WarningsPluginTest#should_override_tab_name_with_groovy_parser`

**Comment by olivergondza** [comment]
Right, the build was aborted prematurely so Warnings tests have not run at all.

File: src/test/java/plugins/WarningsPluginTest.java [hunk 1]
```diff
@@ -86,11 +86,43 @@
...
+            + "String all = matcher.group(1);\n"
+            + "Warning warning = new Warning(all, 42, all, all, all);\n"
+            + "warning.setPackageName('DEAD_' + all);\n"
+            + "warning.setOverridePackageCategoryName('BEEF');\n"
+            + "return warning;";
+
+    /**
+     * Checks that a dynamic Groovy parser can override package tab title.
+     */
+    @Test
```
Comment (uhafner) [inline src/test/java/plugins/WarningsPluginTest.java]: Would be good to have a `@WithPlugin("analysis-core@1.99") `annotation.

**Comment by uhafner** [comment]
Otherwise, the changes look good to me. 

Since the changes are reverted: is a new PR required?

**Comment by olivergondza** [comment]
Assigning to myself to integrate once the review comment is addressed.

**Comment by uhafner** [comment]
I think the comment is already addressed in #343

**Comment by olivergondza** [comment]
Am I right to assume this was replaced by #347?

**Comment by uhafner** [comment]
Sorry, yes. The reference to #343 was a typo...

**Pull request merged**